standard/types.md

```diff
@@ -705,3 +714,184 @@ An *unmanaged_type* is any type that isn’t a *reference_type*, a *type_paramet
+
+- For any reference type `T`, the annotation `?` in `T?` makes `T?` a nullable type, whereas the unannotated `T` is non-nullable.
+- The compiler uses static flow analysis to determine the null state of any reference variable. When nullable warnings are enabled, a reference variable’s null state (§Nullabilities-And-Null-States) is either *not null*, *maybe null*, or *maybe default* and
+  - A warning shall be generated when a possible null value is assigned to a non-nullable type.
+  - A warning shall be generated if a reference variable that has been determined to be *maybe null*, is dereferenced.
+  - The state of a non-nullable reference variable is *not null* unless the compiler can determine that the variable was assigned to an expression that is *maybe null*.
+  - The state of a nullable reference variable is *maybe null* unless the compiler can determine one of two conditions:
+    1. The variable has been definitely assigned a non-`null` value.
+    1. The variable or expression has been checked against `null` before dereferencing it.
+
+> *Example*:
+>
+> <!-- Example: {template:"code-in-main-without-using", name:"NullableAnnotationContext4", ignoredWarnings:["CS0219"], expectedWarnings:["CS8600","CS8600","CS8602"], expectedException:"NullReferenceException"} -->
+> ```csharp
+> #nullable warnings
+> string? s1 = null;    // OK; ? makes s2 nullable
+> string s2 = "text";   // OK; null-state of s2 is "not null"
+> s2 = null;            // OK; null-state of s2 is "maybe null"
+> char c1 = s2[1];      // Warning; dereference of a possible null; throws NullReferenceException
+> c1 = s2![1];          // The warning is suppressed
+> ```
+>
+> *end example*
+
+### §Nullabilities-And-Null-States Nullabilities and null states
+
+Every expression has one of three ***null state***s:
+
+- *maybe null*: The value of the expression may evaluate to null.
+- *maybe default*: The value of the expression may evaluate to the default value for that type.
+- *not null*: The value of the expression definitely won't evaluate to null.
+
+The null state of an expression is determined by the compiler using static flow analysis.
+
+The ***default null state*** of an expression is determined by its type:
+
+- The default null state of a nullable reference type is null.
+- The default null state of a non-nullable reference type is not null.
+
+A warning might result if a variable of a non-nullable type has a potential `null` value assigned to it. Nullable variables can have `null` values freely assigned to them.
+
+> *Note*: Assigning a maybe null expression to a variable `v` of a non-nullable reference type sets the null state of `v` to maybe null. *end note*
+
+Nullability is determined, as follows:
+
+- A non-nullable value type `S` is always non-nullable
+- A nullable value type `S?` is always nullable
+- A reference type `C` in a disabled nullable annotation context is nullable, however its default null state is *not-null*.
```

Comment:
This is the change for null-oblivious, right?

Reply:
Yes.